src/address_operands.rs

```diff
@@ -112,25 +112,21 @@ fn only_reg_write(
     }
 }
 
-fn reg_and_imm_write(
+fn dest_stack_address(
     vm: &mut VMState,
-    output_op_pos: OutputOperandPosition,
+    dest_pos: OutputOperandPosition,
     opcode: &Opcode,
 ) -> TaggedValue {
-    match output_op_pos {
+    match dest_pos {
         OutputOperandPosition::First => {
             let dst0 = vm.get_register(opcode.dst0_index);
             let offset = opcode.imm1;
-            let res = dst0 + TaggedValue::new_raw_integer(U256::from(offset));
-            vm.set_register(opcode.dst0_index, res);
-            res
+            dst0 + TaggedValue::new_raw_integer(U256::from(offset))
         }
         OutputOperandPosition::Second => {
             let dst1 = vm.get_register(opcode.dst1_index);
             let offset = opcode.imm1;
-            let res = dst1 + TaggedValue::new_raw_integer(U256::from(offset));
-            vm.set_register(opcode.dst1_index, res);
-            res
```

Comment:
This `vm.set_register` is actually needed, since when the operand is the second one it always stores on a register, you can check that when this function is called with input `OutputOperandPosition::Second` it is not stored later, maybe it would be better to return the value and store it where it is called